From 485dd83220fa3f41b979d5f6bc3fe5866f673ca7 Mon Sep 17 00:00:00 2001
From: zhanggaofeng <zhanggaofeng9@huawei.com>
Date: Mon, 11 Nov 2019 14:18:42 +0000
Subject: [PATCH] 8233839-aarch64: missing memory barrier in NewObjectArrayStub
 and NewTypeArrayStub

Summary: aarch64: missing memory barrier in NewObjectArrayStub and NewTypeArrayStub
LLT: org.openjdk.jcstress.tests.defaultValues.arrays.small.plain.StringTest
Patch Type: backport
Bug url: https://bugs.openjdk.java.net/browse/JDK-8233839
---
 hotspot/src/cpu/aarch64/vm/c1_Runtime1_aarch64.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hotspot/src/cpu/aarch64/vm/c1_Runtime1_aarch64.cpp b/hotspot/src/cpu/aarch64/vm/c1_Runtime1_aarch64.cpp
index 20a35432d1..c1e48ac97c 100644
--- a/hotspot/src/cpu/aarch64/vm/c1_Runtime1_aarch64.cpp
+++ b/hotspot/src/cpu/aarch64/vm/c1_Runtime1_aarch64.cpp
@@ -877,6 +877,7 @@ OopMapSet* Runtime1::generate_code_for(StubID id, StubAssembler* sasm) {
           __ sub(arr_size, arr_size, t1);  // body length
           __ add(t1, t1, obj);       // body start
           __ initialize_body(t1, arr_size, 0, t2);
+          __ membar(Assembler::StoreStore);
           __ verify_oop(obj);
 
           __ ret(lr);
@@ -905,6 +906,7 @@ OopMapSet* Runtime1::generate_code_for(StubID id, StubAssembler* sasm) {
           __ sub(arr_size, arr_size, t1);  // body length
           __ add(t1, t1, obj);       // body start
           __ initialize_body(t1, arr_size, 0, t2);
+          __ membar(Assembler::StoreStore);
           __ verify_oop(obj);
 
           __ ret(lr);
-- 
2.12.3

